guacamole/src/main/webapp/app/navigation/services/userPageService.js

```diff
@@ -257,15 +257,23 @@ angular.module('navigation').factory('userPageService', ['$injector',
                 canManageConnections.push(dataSource);
             }
 
-            // Determine whether the current user needs access to the session management UI or view connection history
+            // Determine whether the current user needs access to view connection history
             if (
-                    // A user must be a system administrator to manage sessions
+                    // A user must be a system administrator to view connection records
                     PermissionSet.hasSystemPermission(permissions, PermissionSet.SystemPermissionType.ADMINISTER)
             ) {
-                canManageSessions.push(dataSource);
                 canViewConnectionRecords.push(dataSource);
             }
 
+            // Determine whether the current user needs access to view session management
+            if (
+                    // Permission to manage active sessions.
+                       PermissionSet.hasSystemPermission(permissions,           PermissionSet.SystemPermissionType.ADMINISTER)
+                    || PermissionSet.hasActiveConnectionPermission(permissions, PermissionSet.ObjectPermissionType.DELETE)
```

Comment:
This will result in the session management tab only appearing if at least one killable connection is active. Once that connection is killed, the tab will cease to appear (after any cached permissions have been refreshed).

Is this the intended behavior?

Reply:
That's what I was going for, yes, but I don't really have a strong attraction to that behavior if you think it should act some other way?

Reply:
I'm not sure. Only displaying the tab if manageable connections exist definitely makes sense in theory, but the behavior in practice feels inconsistent to me:

https://youtu.be/yiVlfRtZRK4

What do you think about always displaying that tab?

Reply:
Always displaying it is fine with me.  I actually originally implemented it that way, but swapped over to checking for active connections and permissions.

The only side-effect that I see to this is dealing with the issue of behavior when only a single connection is present and going to the home screen vs. automatically connecting.  In general, this wouldn't be an issue - we can adjust the number of pages it looks at up one since we're adding one more that is there for every user - however, there's a corner-case, here, that should at least be considered, even if we ignore it.

Say the user only has access to a single connection, and that connection is open on another system.  The user wants to log in and kill that session, but when they log in they get put into the connection automatically, disconnecting the one on the other client, which starts the loop of connect/disconnect between clients.

Again, it's a corner-case, but worth at least figuring out what we want to do overall with automatic connections, whether this tab is always displayed, etc.

Reply:
For that case, I think we're relatively stuck until something like the proposed automatic-joining functionality is implemented.

I don't think there should be a disconnect loop, though. IIRC, the recent improvements to RDP error handling will cause Guacamole to not automatically reconnect when the user is kicked off specifically because another user has started a session.

Reply:
Also regarding that corner case, such a connection could be configured to allow only one concurrent user. The user attempting to connect would be disallowed with a warning that the connection is in use. If they wish to kick off the existing session (and have permission to do so), they could then open the menu with Ctrl+Alt+Shift, go to Settings, and manually kill it.

Reply:
Okay, changed so it will always show the session management page under Settings.